RandBLAS/sparse.hh

```diff
@@ -578,7 +559,7 @@ void lskges(
     int64_t n, // op(A) is m-by-n
     int64_t m, // op(S) is d-by-m
     T alpha,
-    SparseSkOp<T> &S0,
+    SparseSkOp<T,RNG> &S0,
```

Comment:
I have a strong preference for the RNG not to appear in this function signature. How can we make sure to present an API to the users that doesn't need this?

Reply:
I see in the unit tests that the generator needn't be specified explicitly. So I think we'll be fine and I just need to understand how the templating works, lol.

Reply:
Agree with you on hiding RNG from the users as much as possible. As you've noticed, the user won't have to specify RNG if they're OK with our default selection, which is currently `Philox4x32`. They can swap `Philox4x32` out fairly easily if desired. It's also easy for us to change the default if we ever decide that `Philox4x32` is not what we want. 

However, one option to remove RNG here would be to template on the sketch operator instead of RNG. This would be similar to what I've done in `print_saso` above. I considered doing it that way and didn't come up with a strong reasons for or against.

Reply:
resolved in 9dbc707a

Reply:
Following up on this. I liked ``SparseSkOp<T> &S0`` because it clearly indicated that ``S0`` had to be a ``SparseSkOp``. Those structs had clearly specified semantics. By contrast, templating by ``SKOP`` makes the precise semantics of ``S0`` ambiguous.

Would you say this is a (more or less) unavoidable consequence of suppressing the presence of RNG from the templating? It's okay if that's the case; I just want to understand the constraints based on how C++ does templating. At the end of the day, the ambiguity of templating by ``SKOP`` can always be addressed with documentation that clearly explains what ``SKOP`` might be. But if you know of other ways to address this ambiguity I'd be interested in hearing about them.

Reply:
It looks like the other way to resolve ambiguity would be to use "constraints" and/or "concepts" from [C++20](https://en.cppreference.com/w/cpp/language/constraints). I'd rather not move to C++20 just because of this. So I'll mark this conversation as resolved.